src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -1120,20 +1122,79 @@ void SpatialPooler::updateDutyCyclesHelper_(vector<Real>& dutyCycles,
 
 void SpatialPooler::updateBoostFactors_()
 {
-  for (UInt i = 0; i < numColumns_; i++)
+  if (maxBoost_ > 1.0)
+  {    
+    if (globalInhibition_)
+    {
+      updateBoostFactorsGlobal_();
+    }
+    else
+    {
+      updateBoostFactorsLocal_();
+    }
+  }
+}
+
+void SpatialPooler::updateBoostFactorsGlobal_()
+{
+  Real targetDensity;
+  if (numActiveColumnsPerInhArea_ > 0)
+  {
+    UInt inhibitionArea = pow((Real) (2 * inhibitionRadius_ + 1),
+                              (Real) columnDimensions_.size());
+    inhibitionArea = min(inhibitionArea, numColumns_);
+    targetDensity = ((Real) numActiveColumnsPerInhArea_) / inhibitionArea;
+    targetDensity = min(targetDensity, (Real) 0.5);
+  }
+  else
+  {
+    targetDensity = localAreaDensity_;
+  }
+
+  for (UInt i = 0; i < numColumns_; ++i)
+  {
+    Real boostFactor = exp(-(activeDutyCycles_[i] - targetDensity)
```

Comment:
And why the `exp`? I understand the subtraction (figure out if duty cycle is higher or lower than target density) and the multiplication (scale the difference from target density from a fraction to the magnitude scale specified by max boost), but I don't understand the `exp` after that.